lib/openhab/dsl/types/type.rb

```diff
@@ -41,6 +41,20 @@ def eql?(other)
           equals(other)
         end
 
+        #
+        # Case equality
+        #
+        # @return [Boolean] if the values are of the same Type
+        #                   or item state of the same type
+        #
+        def ===(other)
+          logger.trace("(#{self.class}) #{self} === #{other} (#{other.class})")
+          other = other.state if other.respond_to?(:state)
+          return false unless is_a?(other.class)
```

Comment:
instead of calling is_a? (you should really be using instance_of? in this case) and calling super, just call `eql?(other)` at this point

Reply:
Thanks for the suggestions. I'll implement them in the next push.